src/server/pfs/server/driver.go

```diff
@@ -1060,6 +1060,28 @@ func (d *driver) inspectCommit(ctx context.Context, commit *pfs.Commit, block bo
 	return commitInfo, nil
 }
 
+// resolveCommit is like inspectCommit in that it converts 'commit' (which may
+// be either a commit ID or branch reference) to a repo + commit ID. It accepts
+// an STM so that it can be used in a transaction and avoids an inconsistent
+// call to d.inspectCommit()
+func (d *driver) resolveCommit(stm col.STM, commit *pfs.Commit) error {
+	if len(commit.ID) == uuid.UUIDWithoutDashesLength {
+		return nil // already a commit ID
+	}
```

Comment:
UUIDs also have the property that the 13th character will be `4`, because we're using uuid4 here. We should use that in the check as well.